Tensile/KernelWriterAssembly.py

```diff
@@ -4665,20 +4736,111 @@ def lwaFirstOffset(self, kernel, tP):
       destVgpr = "LocalWriteAddr%s"%tc
 
     dotInterleave = kernel["LocalDotLayout"]
+    WaveOffset = 0
 
     if dotInterleave == 1:
-      kStr += inst("v_mul_u32_u24", \
-          vgpr(destVgpr), \
-          hex(kernel["MacroTile%s"%tP["tensorChar"]] + kernel["LdsPad%s"%tc]), \
-          vgpr(uReg), \
-          "lw%s%s**(MT%s + PAD)"%(tP["tensorChar"], self.unrollChar, tP["tensorChar"]))
-      kStr += inst("_v_add_lshl_u32", \
-          vgpr(destVgpr), \
-          vgpr(tP["gpr"]["lwoT"]), \
-          vgpr(destVgpr), \
-          hex(log2(tP["bpe"])), \
-          "lwFO%s = (lw%s%s + lw%s%s*(MT%s+PAD))*bpe" \
-          % (tc, tc, tc, tc, self.unrollChar, tP["tileChar"]) )
+      if kernel["TransposeLDS"]:
+        # check Transpose case  TLUA=0 TLUB=0
+        # TODO clean-up comment section of assembly code
+        # use different write address pattern
+        sgprLocalWriteoffSet = self.getTmpSgpr(1)
+        tmpVgpr = self.vgprPool.checkOut(1, "tmpVgpr", self.preventVgprOverflowDuringNewTile)
+        tmpVgpr1 = self.vgprPool.checkOut(1, "tmpVgpr1", self.preventVgprOverflowDuringNewTile)
+        wavefronts = kernel["NumThreads"] // globalParameters["WavefrontWidth"]
+        # use  parameter LdsBlockSizePerPAd parameter for adding pad size for lds  fetch efficiency
+        # calculate number of block chunks  and add ldsPAd
+        if not kernel["DirectToLds%s"%tP["tensorChar"]] :
+          LdsPadCnt = (((kernel["MacroTile%s"%tP["tensorChar"]] * kernel["DepthU"])*tP["bpe"])//wavefronts)//kernel["LdsBlockSizePerPad"]
+          WaveOffset = (((kernel["MacroTile%s"%tP["tensorChar"]] * kernel["DepthU"])*tP["bpe"])//wavefronts)
+        else:
+          #DirectToLds each global fetch 256 bytes, there is no vector Offset address for Writing into LDS
+          #m0 + instOffset is writeOffset; padding only accurs at 256 boundary
+          LdsPadCnt = 1
+          WaveOffset = 256
```

Comment:
Always 256? Should this be based on parameters?

Reply:
for DTL; offset is always 256